src/frontend/src/components/dappList.ts

```diff
@@ -0,0 +1,107 @@
+import { html, TemplateResult } from "lit-html";
+
+export type DappDescription = {
+  id: string;
+  name: string;
+  oneLiner?: string;
+  link: string;
+  derivationOrigin?: string;
+  description: string;
+  logo: string;
+};
+
+/**
+ * The marquee is a list of dapps that scrolls horizontally.
+ * It uses a CSS keyframe animation to scroll the each row alternating from left to right and viceversa.
+ * To give the illusion of an infinite scrolling, the list is duplicated horizontally.
+ * when the animation reaches the end of the list, it is reset to the beginning. (default keyframe animation behaviour)
+ */
+
+const createMarqueeList = (dapps: DappDescription[]): TemplateResult => {
+  const itemsPerRow = 5;
+  const totalRows = 4;
+
+  const rows = [];
+
+  for (let i = 0; i < totalRows; i++) {
```

Comment:
I needed an array per row. Would you use something like `new Array(totalRows).fill([])` instead? I thought it was more readable this way.

Reply:
The new implementation looks better because immutable. Does it work out too if the number of dapps does not exactly feel the rows?

Like 4 apps for 2 rows

> dapp0 dapp1 dapp2
> dapp3 dapp0 dapp1

Reply:
Would not be against a test ;)

Reply:
@peterpeterparker yes I also added a comment that explains how it works. (thats what the modulo is doing)